hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -229,7 +229,11 @@ object HoodieSparkSqlWriter {
             }
             sparkContext.getConf.registerAvroSchemas(schema)
             log.info(s"Registered avro schema : ${schema.toString(true)}")
-
+            val columnSet = df.columns.toSet
+            keyGenerator.getRecordKeyFieldNames.foreach(fieldName => if(!columnSet.contains(fieldName)) {
```

Comment:
do all keygenerator return some valid key field names? including  custom ones implemented by users outside the project? If not, then May need to be more defensive in the check.

Reply:
yeah. we can't control user defined ones. if you feel this is too tight of a constraint, then probably we have to drop this patch. 
One option is, to add validateKeyGenProps api to our keyGenerator interface and have it empty for base impl. 
only for internal implementations, we can add this validation.